packages/grid/_modules_/grid/hooks/features/focus/useGridFocus.ts

```diff
@@ -172,17 +190,24 @@ export const useGridFocus = (apiRef: GridApiRef, props: Pick<GridComponentProps,
 
   React.useEffect(() => {
     const doc = ownerDocument(apiRef.current.rootElementRef!.current as HTMLElement);
-    doc.addEventListener('click', handleDocumentClick, true);
+    const handleDocumentClickAsync = (event) => {
+      // If the state is changed during the capture phase, onChange will not be fired
+      // By delaying the state update, there is time for other callbacks to be called first
+      // See https://github.com/facebook/react/issues/13424
```

Comment:
> We rarely use event listeners in the capture phase in the core components. I'm always suspicious when we use them. So, I tried without, I couldn't help myself. The CI is green, the demos seems to work. What am I missing?

True, we don't need to add the listener in the capture phase now. Probably I did this because the code from this PR is actually from #2098, which at the time it was opened #2179 was not merged yet, so stopping the propagation in a `onCellClick` would also break the focus. 

> Off-topic. I personally find the current approach a bit backward. We set the focus state, then the state calls .focus().

The tricky part for me is the Portal support. At the beginning we were doing this approach but we inverted.